hbase-common/src/main/java/org/apache/hadoop/hbase/io/crypto/Encryption.java

```diff
@@ -640,20 +640,17 @@ public static void incrementIv(byte[] iv) {
   }
 
   public static void incrementIv(byte[] iv, int v) {
+    // v should be > 0
     int length = iv.length;
-    boolean carry = true;
-    // TODO: Optimize for v > 1, e.g. 16, 32
-    do {
-      for (int i = 0; i < length; i++) {
-        if (carry) {
-          iv[i] = (byte) ((iv[i] + 1) & 0xFF);
-          carry = 0 == iv[i];
-        } else {
-          break;
-        }
+    int sum = 0;
+    for (int i = 0; i < length; i++) {
+      if (v <= 0) {
+        break;
       }
-      v--;
-    } while (v > 0);
+      sum = v + iv[i];
```

Comment:
What if iv[i] is negative here? In a first thought I think we should consider each byte as unsigned?

Reply:
Right, we should care the negative here. Has refined the logic here and added negative UT cases.